File: practice-algos/missing_number/gpt_sol.py
def missingNumber(nums):
    n = len(nums)
    expected_sum = n * (n + 1) // 2
    actual_sum = sum(nums)
    return expected_sum - actual_sum

# Test cases
print(missingNumber([3,0,1]))  # Output: 2
print(missingNumber([0,1]))    # Output: 2
print(missingNumber([9,6,4,2,3,5,7,0,1]))  # Output: 8

File: practice-algos/missing_number/gpt_sol.js
function missingNumber(nums) {
    const n = nums.length;
    const expectedSum = n * (n + 1) / 2;
    const actualSum = nums.reduce((acc, num) => acc + num, 0);
    return expectedSum - actualSum;
}

// Test cases
console.log(missingNumber([3,0,1]));  // Output: 2
console.log(missingNumber([0,1]));    // Output: 2
console.log(missingNumber([9,6,4,2,3,5,7,0,1]));  // Output: 8
